eth/handler.go

```diff
@@ -399,3 +416,49 @@ func (h *handler) txBroadcastLoop() {
 		}
 	}
 }
+
+// missingBodyLoop listens to the MissingBody event in Slice and calls the blockAnnounces.
+func (h *handler) missingBodyLoop() {
+	defer h.wg.Done()
+	for {
+		select {
+		case header := <-h.missingBodyCh:
+
+			// Get the min(sqrt(len(peers)), minPeerRequest)
+			var peerThreshold int
+			sqrtNumPeers := int(math.Sqrt(float64(len(h.peers.peers))))
+			if sqrtNumPeers < minPeerRequest {
+				peerThreshold = len(h.peers.peers)
```

Comment:
if `minPeerRequest` is `3`, but `len(peers)` is `4`, the sqrt would be `2`. You should set it to 3 (i.e. `minPeerRequest`) in this case, but instead you set it to 4 (i.e. `len(peers)`).

Reply:
I think line 431 should be:
```peerThreshold = min(minPeerRequest, len(h.peers.peers))```

Reply:
oops, have to set it to the minPeer. good catch.